### PR TITLE
Enhance CI checks

### PR DESCRIPTION
**Pull request by doomspork**
Opening as a draft to socialize and solicit feedback before calling this "done".

This PR enhances CI checks in the following ways:

- Break out Format from Test ✅ 
- Add Credo ✅ 
- Add Dialyzer ✅ 
- Add Caching
- Expand version matrix (@davydog187 is there is a list of versions you want to shoot to support beyond just the rc?) ✅ 

Outstanding issues preventing Credo from passing:

**Credo**

@davydog187 how do you want to handle existing TODO, convert to issues?

image

We can add all these existing issues to an ignore for now.

### CODE REVIEW

**Comment by doomspork** [comment]
The dialyzer error might be highlighting a bug. I threw together a quick unit test for `migration.ex` to cover this example from the docs: `SELECT create_hypertable('conditions', 'time', chunk_time_interval => 86400000000)` and I get an argument error.

Will investigate and try to include a fix here.

File: lib/timescale/migration.ex [hunk 1]
```diff
@@ -67,9 +67,8 @@ defmodule Timescale.Migration do
   defp opts_to_sql_arg([]), do: ""
 
   defp opts_to_sql_arg(opts) do
-    ", " <>
-      Enum.map(opts, fn {k, v} ->
-        "#{k} => #{v}"
-      end)
+    Enum.map_join(opts, ", ", fn {k, v} ->
+      "#{k} => #{v}"
+    end)
```
Comment (doomspork) [inline lib/timescale/migration.ex:72]: I'd like to figure out a clean way to get test coverage for the `migration.ex` file 🤔

**Comment by davydog187** [comment]
Yes, let's convert the todos to issues and remove from the codebase

File: lib/timescale/migration.ex [hunk 2]
```diff
@@ -67,9 +67,8 @@ defmodule Timescale.Migration do
   defp opts_to_sql_arg([]), do: ""
 
   defp opts_to_sql_arg(opts) do
-    ", " <>
-      Enum.map(opts, fn {k, v} ->
-        "#{k} => #{v}"
-      end)
+    Enum.map_join(opts, ", ", fn {k, v} ->
```
Comment (davydog187) [inline lib/timescale/migration.ex:70]: DId Credo suggest this refactor? Nice!

Comment (doomspork) [reply lib/timescale/migration.ex:67]: This is actually a bug caught by Dialyzer @davydog187! I tested it out locally and sure enough the prior version resulted in an argument error.

**Comment by doomspork** [comment]
All checks pass and caching appears to be working well

File: lib/timescale/migration.ex (hunk 2 above)
Comment (davydog187) [reply lib/timescale/migration.ex:67]: Amazing!

**Review by davydog187 — APPROVED** [review]
This is great, thank you @doomspork 🥳

**Pull request merged**